react-app-drawer-component/src/app-drawer.jsx

```diff
@@ -0,0 +1,57 @@
+/* eslint-disable prefer-const */
+import React, { useState } from 'react';
+
+export default function AppDrawer() {
+  const [iconClicked, setIconClicked] = useState(false);
+  const [image, setImage] = useState('https://i.imgur.com/7UWSnsD_d.webp?maxwidth=760&fidelity=grand');
+  const vaporeon = 'https://i.imgur.com/7UWSnsD_d.webp?maxwidth=760&fidelity=grand';
+  const gardevoir = 'https://i.imgur.com/qeAZRuq.png';
+  const lopunny = 'https://i.imgur.com/kerbwps.png';
+
+  const handleClick = (e) => {
+    setIconClicked(!iconClicked);
+    switch (e.currentTarget.id) {
```

Comment:
Ah, this is how you detect which menu item was clicked. See if you can do it in straight React without examining the DOM.